great_expectations/core/batch_definition.py

```diff
@@ -17,32 +17,36 @@
     )
     from great_expectations.datasource.fluent.interfaces import Batch, DataAsset
 
+PartitionerT = TypeVar("PartitionerT", bound=Optional[Partitioner])
```

Comment:
Can you remove `Optional` from the `bound` argument? It looks like you often use `Optional[Partitioner]` when this can be `None`. I'm not sure if this works but maybe you can define `class BatchRequest(pydantic.GenericModel, Generic[Optional[PartitionerT]])` and `class BatchDefinition(pydantic.GenericModel, Generic[Optional[PartitionerT]])`?

Reply:
thanks @billdirks! I gave this some more thought, and have updated `PartitionerT` from a single upper bound of `Partitioner | None` to multiple upper bounds of `Partitioner` and `None` (and will soon update this to use the upper bounds of `SqlPartitioner`, `RegexPartitioner`, and `None`. Classes using the type can choose the upper bound that suites their use case, and then decide whether or not to make the field optional where its used. I guess its a subtle distinction, but hopefully make it clearer from reading the type that `PartitionerT` has two distinct valid types.